torch_xla/experimental/distributed_checkpoint/manager.py

```diff
@@ -105,6 +109,27 @@ def __init__(self,
     self.save_period = save_period
     self.max_to_keep = max_to_keep
 
+    self._async_queue = queue.Queue(maxsize=async_queue_size)
+    self._chkpt_thread = threading.Thread(target=self._async_worker, daemon=True)
+    self._chkpt_thread.start()
+
+    # Create a CPU process group to coordinate the checkpoint.
+    self.pg = dist.new_group(backend='gloo')
```

Comment:
I thought you suggested that the user should do this?

Reply:
A few questions:
- does this backend always work? `gloo` can be used not only for CPU, but also GPU. If you don't specify the members, does it only contain the CPU processes?
- how doe this differ from the default gloo pg?

Reply:
The user will need to initialize the default process group, but since the default group is thread-local, we need to create a handle on a subgroup for use in the background thread.

To @yeounoh:
- This will inherit the ranks from the default group. With SPMD on TPUs, this has to be a CPU group, but with GPU it could be on-device. Probably the right approach is to not specify `gloo` here and just inherit the default group's backend.
- The resulting group should be identical to the default group.

Reply:
Now I missed the [event loop](https://docs.python.org/3/library/asyncio-eventloop.html) in any other modern programming languages...

Reply:
@jonb377 good, can you just add a TODO comment, that need to verify for XLA:GPU